libs/host/GarnetServer.cs

```diff
@@ -76,7 +77,7 @@ public class GarnetServer : IDisposable
         /// </summary>
         /// <param name="commandLineArgs">Command line arguments</param>
         /// <param name="loggerFactory">Logger factory</param>
-        public GarnetServer(string[] commandLineArgs, ILoggerFactory loggerFactory = null, bool cleanupDir = false)
+        public GarnetServer(string[] commandLineArgs, IAuthenticationSettings authenticationSettings = null, ILoggerFactory loggerFactory = null, bool cleanupDir = false)
```

Comment:
Rename to authenticationSettingsOverride and use that if not null, instead of creating a new enum type.